client/src/client.rs

```diff
@@ -200,4 +201,16 @@ impl Client {
                 info.fallback_to_local_pow
             })
     }
+
+    /// Validates if a bech32 HRP matches the one from the connected network.
+    pub async fn validate_bech32_hrp(&self, bech32_hrp: &str) -> Result<()> {
```

Comment:
this method name sounds like it could be used to validate if an hrp is valid, not if its for the current network.
i know the comment says that, but maybe we rename to `bech32_hrp_matches` and return Result<bool> instead?
A whole error thrown seems excessive to me

Reply:
hmm it does make the code below less readable having to do a `if matches {}`
Lets leave the result as is

Reply:
Renamed